File: core/Core/Models/Deposit.cs
using System;
using System.Collections.Generic;
using DateOnly = global::System.DateOnly;

namespace Core.Models;

public partial class Deposit
{
    public int DepositId { get; set; }

    public required int CustomerId { get; set; }

    public required decimal Amount { get; set; }

    public required decimal InterestRate { get; set; }

    public required System.DateOnly StartDate { get; set; }

    public System.DateOnly? EndDate { get; set; }

    public required string Status { get; set; }

    public required Customer Customer { get; set; }
}

File: analytic/domain/deposit/deposit.py
from datetime import date
from decimal import Decimal


class Deposit:
    def __init__(self,
                 deposit_id: int,
                 customer_id: int | None,
                 amount: Decimal | None,
                 interest_rate: Decimal | None,
                 start_date: date | None,
                 end_date: date | None,
                 status: str | None):
        self.deposit_id = deposit_id
        self.customer_id = customer_id
        self.amount = amount
        self.interest_rate = interest_rate
        self.start_date = start_date
        self.end_date = end_date
        self.status = status
